server/controllers/profileController.js

```diff
@@ -0,0 +1,100 @@
+const Profile = require("../models/Profile");
+const { validationResult } = require("express-validator");
+
+module.exports.getProfileList = async (req, res, next) => {
+  try {
+    const profiles = await Profile.find();
+    if (!profiles) {
+      return res.status(404).json({ err: "No profiles founds" });
+    }
+    res.status(200).json(profiles);
+  } catch (err) {
+    res.status(400).json("Unable to retreive profiles");
+  }
+};
+
+module.exports.getProfileById = async (req, res, next) => {
+  try {
+    const profile = await Profile.findById(req.params.id);
+    if (!profile) {
+      return res.status(404).json({ msg: "Profile not found" });
+    }
+    res.status(200).json(profile);
+  } catch (err) {
+    res.status(400).json("Unable to retreive profile");
+  }
+};
+
+module.exports.updateProfile = async (req, res, next) => {
+  const data = {
```

Comment:
maybe saving the attributes in the `req.body` object in a `const` would allow it easier to manage
```
const {firstName, lastName, gender, birthDate} = req.body;

const data = {
  firstName: firstName
  ...
}
```

basically you don't have to type out `req.body` every time - easier to read and maintain :)

Reply:
you are right! thanks